network/src/protocols/disconnect_message.rs

```diff
@@ -63,3 +79,33 @@ impl ServiceProtocol for DisconnectMessageProtocol {
         });
     }
 }
+
+impl DisconnectMessageProtocol {
+    // parse disconnect message and return peer_ip, ban seconds and reason(if exists)
+    // refer to NetworkState::ban_session() for message format
+    fn parse_disconnect_message(
+        &self,
+        session_id: SessionId,
+        message: &str,
+    ) -> (String, String, String) {
+        let addr = self.0.with_peer_registry(|reg| {
+            reg.get_peer(session_id)
+                .filter(|peer| !peer.is_whitelist)
```

Comment:
why filter out whitelist?

Reply:
filter of whitelist is unnecessary and removed...